app/assets/javascripts/oxalis/view/nml_upload_zone_container.js

```diff
@@ -55,13 +56,19 @@ function OverlayDropZone({ children }) {
   );
 }
 
-function NmlDropArea({ showClickHint }) {
+function NmlDropArea({ showClickHint, isAllowed }) {
   return (
     <React.Fragment>
       <div>
         <Icon type="inbox" style={{ fontSize: 180, color: "rgb(58, 144, 255)" }} />
       </div>
-      <h5>Drop NML files here{showClickHint ? " or click to select files" : null}...</h5>
+      {isAllowed ? (
+        <h5>Drop NML files here{showClickHint ? " or click to select files" : null}...</h5>
+      ) : (
+        <h5 style={{ color: "rgb(247,80,61)" }}>
+          Upload of NML files is not allowed. Please check your permissions.
```

Comment:
The message is a tad harsh. Even admins would see the message if they open a tracing (or dataset) in view mode. So, the word "permissions" is not ideal, in my opinion. I'd change it to:

"You cannot upload NML files into a read-only tracing".